### ISSUE TITLE
Routing fails for reverse proxy installation at the domain root

### ISSUE THREAD

**Issue by lukasbestle** [Kozea/Radicale#900]
**Description of the bug**:

I'm using Radicale behind Apache as a reverse-proxy. Because I don't have access to `mod_proxy`, I need to use `mod_rewrite`'s `[P]` flag to proxy the requests (see the code snippet below).

If I run Radicale in a subdirectory (like `/radicale` in the [reverse-proxy docs](https://radicale.org/proxy/)) and set the `X-Script-Name` header accordingly, everything works perfectly (without any changes to the Radicale config).

Also it all works perfectly if I access the Radicale server directly at `http://localhost:12345`.

But in my setup at the root of the domain, the routing in the Radicale web interface doesn't work:

- If I access `https://example.com/`, I get "Access to the requested resource forbidden.".
- If I access `https://example.com/.web/`, I can access the web interface login page, but can't login (I get `Error: 403` below the form).

Is there anything wrong with the header I'm setting from Apache? Is there a special value for "installed at the root of the domain"? Or is this a known issue?

Thanks for your time!

---

**Used software and versions**:

- Radicale 2.1.11 installed via `pip`
- Python 3.6.5
- Apache 2.4.6 with `mod_rewrite`

**Radicale config**:

```ini
[server]
hosts = 0.0.0.0:12345

[auth]
type = htpasswd
htpasswd_filename = /home/my-user/radicale/users
htpasswd_encryption = bcrypt

[storage]
filesystem_folder = /home/my-user/radicale/storage
```

**Apache configuration (in `.htaccess`)**:

```apache
RewriteEngine On

RewriteRule ^(.*) http://localhost:12345/$1 [P]
RequestHeader set X-Script-Name /
```

**Comment by mookie-** [comment]
Hello, could you please add the apache error.log entrys to this issue? I think it would be easier to help you with that.

**Comment by lukasbestle** [comment]
Thank you for your reply!

As there is no error on the Apache level, there is nothing in my `error_log`. Unfortunately I don’t have access to the Apache configuration and therefore can‘t increase the verbosity level.

Is my setup generally correct? Please note that the app itself works (my CalDAV clients can connect perfectly). It‘s just the home route and the web interface.

**Comment by mookie-** [comment]
@lukasbestle I'm sorry for the late response. I've installed radicale and apache on my local machine and used your configuration. Unfortunately I can't reproduce your problem. It's working fine (I've only tested the web access).

To help you with that without an error log, I would probably need the whole apache configuration.

**Comment by lukasbestle** [comment]
Hm, strange.

Unfortunately I don't have any access to the Apache configuration as I already wrote (shared hosting), so I'm afraid I can't contribute any further information. We'll have to see if anyone else has the same problem.

**Comment by pbiering** [comment]
did you manage to get this proper fixed with lastest 3.1.x?

**Comment by lukasbestle** [comment]
Can confirm that this no longer happens with v3.
